src/extension_custom_scripts.c

```diff
@@ -0,0 +1,157 @@
+#include "extension_custom_scripts.h"
+
+// Prevent recursively running custom scripts
+static bool running_custom_script = false;
```

Comment:
I find this behavior mysterious. Can we cover this by tests so it can be fully understood?

I can help by adding the tests on another PR too, if it requires some other extension it's fine (I've added tests requiring [plpgsql_check](https://github.com/supabase/supautils/pull/129) and [pgmq](https://github.com/supabase/supautils/pull/134/files) before). Just let me know how can I reproduce.

Reply:
The pathological behavior we're trying to avoid is an infinite loop caused by a `create extension` triggering a custom script that then runs a `create extension`, which in turn triggers a custom script, etc.

Can't really put the infinite loop as a test case, but preventing nested custom scripts can be done with 2 extensions: have custom scripts for both where one does a `create extension` for the other, and have the other one do a side effect (e.g. create a table with some sentinel name) and we assert not to happen.